### PR TITLE
adding new api methods

### PR DESCRIPTION
**Pull request by stevengill**
###  Summary

Issue: #1091 

Added the following API methods:

- [admin.conversations.archive](https://api.slack.com/methods/admin.conversations.archive)
- [admin.conversations.convertToPrivate](https://api.slack.com/methods/admin.conversations.convertToPrivate)
- [admin.conversations.create](https://api.slack.com/methods/admin.conversations.create)
- [admin.conversations.delete](https://api.slack.com/methods/admin.conversations.delete)
- [admin.conversations.disconnectShared](https://api.slack.com/methods/admin.conversations.disconnectShared)
- [admin.conversations.ekm.listOriginalConnectedChannelInfo](https://api.slack.com/methods/admin.conversations.ekm.listOriginalConnectedChannelInfo)
- [admin.conversations.getConversationPrefs](https://api.slack.com/methods/admin.conversations.getConversationPrefs)
- [admin.conversations.getTeams](https://api.slack.com/methods/admin.conversations.getTeams)
- [admin.conversations.invite](https://api.slack.com/methods/admin.conversations.invite)
- [admin.conversations.rename](https://api.slack.com/methods/admin.conversations.rename)
- [admin.conversations.search](https://api.slack.com/methods/admin.conversations.search)
- [admin.conversations.setConversationPrefs](https://api.slack.com/methods/admin.conversations.setConversationPrefs)
- [admin.conversations.unarchive](https://api.slack.com/methods/admin.conversations.unarchive)
- [admin.emoji.add](https://api.slack.com/methods/admin.emoji.add)
- [admin.emoji.addAlias](https://api.slack.com/methods/admin.emoji.addAlias)
- [admin.emoji.list](https://api.slack.com/methods/admin.emoji.list)
- [admin.emoji.remove](https://api.slack.com/methods/admin.emoji.remove)
- [admin.emoji.rename](https://api.slack.com/methods/admin.emoji.rename)
- [admin.users.session.invalidate](https://api.slack.com/methods/admin.users.session.invalidate)
- [apps.event.authorizations.list](https://api.slack.com/methods/apps.event.authorizations.list)

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/web-api/src/methods.ts [hunk 1]
```diff
@@ -471,30 +524,97 @@ export interface AdminAppsRestrictedListArguments extends WebAPICallOptions, Tok
...
+export interface AdminConversationsGetConversationPrefsArguments extends WebAPICallOptions, TokenOverridable {
+  channel_id: string;
+}
+export interface AdminConversationsGetTeamsArguments
+  extends WebAPICallOptions, TokenOverridable, CursorPaginationEnabled {
+  channel_id: string;
+}
+export interface AdminConversationsInviteArguments extends WebAPICallOptions, TokenOverridable {
+  channel_id: string;
+  user_ids: string | string[];
```
Comment (stevengill) [inline packages/web-api/src/methods.ts:561]: Some of these I'm not sure based on the description if it takes either a `string` or a `string[]` or just one of those. @seratch is there a place in webapp that I can confirm these type of properties?

Comment (seratch) [reply packages/web-api/src/methods.ts:561]: @stevengill Even when passing a `string[]` value here, the value will be sent as a single string formatted by `JSON.stringify` in a form request body. https://github.com/slackapi/node-slack-sdk/blob/%40slack/web-api%405.12.0/packages/web-api/src/WebClient.ts#L385-L388

In other SDKs, we always generate a comma-separated format string value (not JSON string) from a list/array data. As far as I know, that's the safest way as the format is the only one mentioned in the API documentation. https://api.slack.com/methods/admin.conversations.invite

That said, doing `JSON.stringify` should always work as well. We've never encountered any issues with the current approach of this SDK.

File: packages/web-api/src/methods.ts [hunk 2]
```diff
@@ -471,30 +524,97 @@ export interface AdminAppsRestrictedListArguments extends WebAPICallOptions, Tok
...
   team_id?: string;
   target_team_ids?: string[];
   org_channel?: boolean;
 }
+export interface AdminConversationsUnarchiveArguments extends WebAPICallOptions, TokenOverridable {
+  channel_id: string;
+}
+export interface AdminEmojiAddArguments extends WebAPICallOptions, TokenOverridable {
+  name: string;
+  alias_for: string;
```
Comment (seratch) [inline packages/web-api/src/methods.ts:604]:
```suggestion
  url: string;
```

File: packages/web-api/src/methods.ts [hunk 3]
```diff
@@ -471,30 +524,97 @@ export interface AdminAppsRestrictedListArguments extends WebAPICallOptions, Tok
...
+}
+export interface AdminEmojiAddAliasArguments extends WebAPICallOptions, TokenOverridable {
+  name: string;
+  url: string;
```
Comment (seratch) [inline packages/web-api/src/methods.ts:608]:
```suggestion
  alias_for: string;
```

File: packages/web-api/src/methods.ts [hunk 4]
```diff
@@ -634,6 +758,25 @@ export interface APITestArguments extends WebAPICallOptions {}
   /*
    * `apps.*`
    */
+export interface AppsEventAuthorizationsListArguments
+  extends WebAPICallOptions, TokenOverridable, CursorPaginationEnabled {
+  event_context: string;
+}
+export interface AppsPermissionsInfoArguments extends WebAPICallOptions, TokenOverridable {}
```
Comment (seratch) [inline packages/web-api/src/methods.ts:765]: It's totally fine to have these but just for your information, `apps.permissions.*` are for Workspace Apps. So, no new developer will be using these ones.

**Review by seratch — CHANGES_REQUESTED** [review]
`admin.emoji.*` need fixes.

File: packages/web-api/src/methods.ts (hunk 2 above)
Comment (stevengill) [reply packages/web-api/src/methods.ts:604]: Nice catch!

**Comment by stevengill** [comment]
@seratch take a look when you got a minute

**Review by seratch — APPROVED** [review]

**Comment by boblauer** [comment]
Is there an eta on when these changes will be published to npm?

**Comment by stevengill** [comment]
@boblauer Doing a release today!

**Comment by mwbrooks** [comment]
@boblauer published as [@slack/web-api@5.13.0](https://github.com/slackapi/node-slack-sdk/releases/tag/%40slack%2Fweb-api%405.13.0)!

**Pull request merged**